packages/profiling-node/bindings/cpu_profiler.cc

```diff
@@ -604,8 +629,15 @@ static void GetSamples(const napi_env &env, const v8::CpuProfile *profile,
       }
     }
 
-    napi_value sample = CreateSample(
-        env, stack_index, sample_timestamp - profile_start_time_us, thread_id);
+    uint64_t sample_delta_us = sample_timestamp - profile_start_time_us;
+    uint64_t sample_timestamp_ns = sample_delta_us * 1e3;
+    double seconds_since_start =
```

Comment:
Do we need to validate these timestamps?

Reply:
I could validate that they fall into the range of the profile, but I dont think it is necessary. Worst case, they will get rendered weirdly and we'll catch it there. It's also an extremely hot path as we might have a lot of samples here so I'd rather defer this to the backend